evm_arithmetization/src/generation/linked_list.rs

```diff
@@ -66,7 +82,7 @@ impl<'a, const N: usize, T: LinkedListType> LinkedList<'a, N, T> {
         mem: &'a [Option<U256>],
         segment: Segment,
     ) -> Result<Self, ProgramError> {
-        if mem.is_empty() {
+        if mem.is_empty() || mem.len() % N != 0 {
```

Comment:
I'm actually curious about the `mem.is_empty()` case. This shouldn't ever happen, no?

Reply:
It happens on the first cycles, when we still haven reached `%init_access_lists`.

Reply:
Hmm ok, but these should never be accessed before the initialization anyway, I find it a bit confusing

Reply:
For all N>0 . 0 % N == 0, so the RHS of the disjunction should suffice.

Reply:
I removed the `mem.is_empty()`